lib/r10k/module/definition.rb

```diff
@@ -0,0 +1,56 @@
+require 'r10k/module'
+
+class R10K::Module::Definition < R10K::Module::Base
+
+  attr_reader :version
+
+  def initialize(name, dirname:, args:, implementation:, environment: nil)
+    @original_name  = name
+    @original_args  = args.dup
+    @implementation = implementation
+    @version        = implementation.statically_defined_version(name, args)
+
+    super(name, dirname, args, environment)
+  end
+
+  def to_implementation
+    mod = @implementation.new(@title, @dirname, @original_args, @environment)
+
+    mod.origin = origin
+    mod.spec_deletable = spec_deletable
+
+    mod
+  end
+
+  # syncing is a noop for Stub modules
+  def sync(args = {})
+    logger.debug1(_("Not updating module %{name}, assuming content unchanged") % {name: name})
+  end
+
+  def status
+    :insync
+  end
+
+  def properties
+    type = nil
+
+    if @args[:type]
+      type = @args[:type]
+    elsif @args[:ref] || @args[:commit] || @args[:branch] || @args[:tag]
+      type = 'git'
+    elsif @args[:svn]
+      # This logic is clear and included for completeness sake, though at
+      # this time Stub modules do not support SVN versions.
+      type = 'svn'
+    else
+      type = 'forge'
+    end
+
+    {
+      expected: version,
+      actual: 'assumed',
```

Comment:
Yes, I can document that.

FWIW, we don't use this info directly in code-manager, though we serve it via an endpoint `deploys/data`. We don't document that endpoint publicly, but CD4PE does use it as part of IA. This change should match the same "type", ie this field for modules in general is an arbitrary string. However, I'm confirming with CD4PE that this won't have negative affects on IA, because to get the "actual" value would require resolving the ref, which will do a fetch, which is the thing we're trying to avoid.